### PR TITLE
update --strip-metadata for new debug format

### PR DESCRIPTION
**Pull request by JeffBezanson**
It would be nice to be able to remove the DebugInfo objects themselves, but this gets the vast majority of the bytes, and removing those objects I think would require adding lots of checks in various places so this is good for now.

### CODE REVIEW

**Review by vtjnash — COMMENTED** [review]
FWIW, this may be unsound, technically, since they are immutable objects

**Comment by JeffBezanson** [comment]
Good point. I think it would be ok to just make them mutable then?

**Comment by vtjnash** [comment]
Not ideal to mutate them ever though, since we reference them from many places

**Comment by vtjnash** [comment]
And we rely on them not changing for efficient ji files

**Comment by JeffBezanson** [comment]
How do we rely on that?

But anyway, are you saying you'd prefer to make the debuginfo fields `Union{Nothing, DebugInfo}`? That would be fine since it would save more space anyway.

**Comment by vtjnash** [comment]
We can avoid copying them into the caller, since we know they cannot change. You could grab something like `Serialization.NullDebugInfo` and use that everywhere?

File: src/staticdata.c [hunk 1]
```diff
@@ -2384,6 +2387,7 @@ static void strip_specializations_(jl_method_instance_t *mi)
                 if (jl_atomic_cmpswap_relaxed(&codeinst->inferred, &inferred, stripped)) {
                     jl_gc_wb(codeinst, stripped);
                 }
+                record_field_change((jl_value_t**)&codeinst->debuginfo, (jl_value_t*)jl_nulldebuginfo);
             }
         }
```
Comment (topolarity) [inline src/staticdata.c:2392]:
```suggestion
            }
        }
        if (jl_options.strip_metadata)
            record_field_change((jl_value_t**)&codeinst->debuginfo, (jl_value_t*)jl_nulldebuginfo);
```

Even if the `inferred` field is nothing, we should still clear out the `debuginfo`

**Pull request merged**